src/parse/asp/exec.go

```diff
@@ -35,6 +40,90 @@ func init() {
 	execPromises = make(map[execKey]*execPromise, initCacheSize)
 }
 
+// doExec fork/exec's a command and returns the output as a string.  exec
+// accepts either a string or a list of commands and arguments.  The output from
+// exec() is memoized by default to prevent side effects and aid in performance
+// of duplicate calls to the same command with the same arguments (e.g. `git
+// rev-parse --short HEAD`).  The output from exec()'ed commands must be
+// reproducible.
+//
+// NOTE: Commands that rely on the current working directory must not be cached.
+func doExec(s *scope, args []pyObject) pyObject {
```

Comment:
Why does doExec() still take args in python types? Is this just an artifact from when exec() was going to be exposed to starlark?

Why not just []string?

Reply:
Cleaned up and am using something closer to idiomatic Go now.